Generate example sdk using gradle

slifty opened the pull request:
This PR sets up a "hello world" generator in the typescript namespace and adds the appropriate gradle configuration for invoking the generator.

It also updates our editorconfig to use tabs, as we're moving other aspects of the project to tabs.

You should be able to test this by following the instructions in the README (ultimately invoking `./gradlew generateSwaggerCode to see the generated content)

Resolves #5

reefdog commented on README.md:
```diff
@@ -1,2 +1,32 @@
+This repository houses the infrastructure for generating PDC SDKs.
+
+We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates
+to make sdk generation easier in particular target languages.  The codegen project supports a [many languages](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources)
```
```suggestion
to make sdk generation easier in particular target languages. The codegen project supports a [many languages](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources)
```

reefdog replied:
(This is not a real suggestion, so much as a poetic lament.)

slifty replied:
I do what prettier tells me to do justin, your wish is granted.

reefdog replied:
😮 Prettier removed the double-spaces?? I LOVE PRETTIER

slifty replied:
It did in the other repo -- https://github.com/PhilanthropyDataCommons/service/pull/695#discussion_r1443550634

(we aren't running it here, but I try to abide by the demand of automated tools in all aspects of life)

reefdog replied:
I love that Prettier has more credibility than _me_.

bickelj commented on README.md:
````diff
@@ -1,2 +1,32 @@
+
+```shell
+> cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
+> envsubst ./configuration/typescript.json.template>./configuration/typescript.json
````
This command appeared to wait for more input on my machine.

slifty replied:
This line should be: `envsubst < ./configuration/typescript.json.template > ./configuration/typescript.json `

bickelj commented on build.gradle:
```diff
@@ -0,0 +1,28 @@
+}
+
+repositories {
+		mavenCentral()
```
Whitespace here seems off.

bickelj reviewed (approved):
This looks way gradleyer than before! Nice!

Minor suggestions above.

slifty commented:
w00t thank you @bickelj!  Addressed the suggestions, and also reorganized the commits to separate the custom template from the initial setup.  Merging in and going forward to actually write custom templates!
